.github/workflows/wash.yml

```diff
@@ -133,3 +153,19 @@ jobs:
 
       - name: Ensure `wash new` works
         run: wash new component ${{ matrix.template.name }} --template-name ${{ matrix.template.name }}
+
+  # This check always runs, and only succeeds if wash tests should run and they pass
+  wash_successful_checks:
+    needs:
+      - should_run
+      - unit_tests
+      - integration_tests
+      - e2e_tests_wash_new_actor
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    steps:
+      # Conditionally fail job depending on results of previous actions
+      - if: ${{ contains(needs.*.result, 'failure') }}
+        run: exit 1
+      - if: ${{ !contains(needs.*.result, 'failure') }}
+        run: exit 0
```

Comment:
Reminds me of https://github.com/linkerd/linkerd2/blob/c463750c6dc497ae4acc0b9bbb2c337bea5e0efd/.github/workflows/integration.yml#L358-L371 👍

Reply:
Oh that's good inspiration